### PR TITLE
create go1.21.9 directory

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind ci

#### What this PR does / why we need it:
Build runc with golang 1.21.

Set up correct path.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by sohankunkerkar** [comment]
Seeing the following error on a Fedora VM with this change:
```sh
TASK [build runc] ********************************************************************************************************************************************
task path: /home/skunkerk/cri-o/contrib/test/ci/build/runc.yml:29
fatal: [localhost]: FAILED! => {
    "changed": false,
    "cmd": "/usr/bin/gmake GO=/usr/local/go1.21.9/go/bin/go",
    "invocation": {
        "module_args": {
            "chdir": "/usr/go/src/github.com/opencontainers/runc",
            "file": null,
            "jobs": null,
            "make": null,
            "params": {
                "GO": "/usr/local/go1.21.9/go/bin/go"
            },
            "target": null,
            "targets": null
        }
    },
    "rc": 2
}

STDOUT:

rm -f libcontainer/dmz/binary/runc-dmz
/usr/local/go1.21.9/go/bin/go generate -tags "seccomp urfave_cli_no_docs " ./libcontainer/dmz
make[1]: Entering directory '/usr/go/src/github.com/opencontainers/runc/libcontainer/dmz'
gcc -fno-asynchronous-unwind-tables -fno-ident -s -Os -nostdlib -lgcc -static -o binary/runc-dmz _dmz.c
strip -gs binary/runc-dmz
make[1]: Leaving directory '/usr/go/src/github.com/opencontainers/runc/libcontainer/dmz'
/usr/local/go1.21.9/go/bin/go build -trimpath "-buildmode=pie"  -tags "seccomp urfave_cli_no_docs " -ldflags "-X main.gitCommit=v1.2.0-rc.1-4-g5e0ec3fb -X main.version=1.2.0-rc.1+dev " -o runc .



STDERR:

# internal/goarch
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/unsafeheader
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/coverage/rtcov
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/cpu
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/godebugs
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/goos
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/goexperiment
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# runtime/internal/atomic
flag provided but not defined: -compiling-runtime
usage: asm [options] file.s ...
Flags:
  -D value
    	predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
    	include directory; can be set multiple times
  -S	print assembly and machine code
  -V	print version and exit
  -d value
    	enable debugging settings; try -d help
  -debug
    	dump instructions as they are parsed
  -dynlink
    	support references to Go symbols defined in other shared libraries
  -e	no limit on number of errors reported
  -gensymabis
    	write symbol ABI information to output file, don't assemble
  -linkshared
    	generate code that will be linked against Go shared libraries
  -o string
    	output file; default foo.o for /a/b/c/foo.s as first argument
  -p string
    	set expected package import to path (default "<unlinkable>")
  -shared
    	generate code that can be linked into a shared library
  -spectre list
    	enable spectre mitigations in list (all, ret)
  -trimpath string
    	remove prefix from recorded source file paths
  -v	print debug output
# runtime/internal/syscall
flag provided but not defined: -compiling-runtime
usage: asm [options] file.s ...
Flags:
  -D value
    	predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
    	include directory; can be set multiple times
  -S	print assembly and machine code
  -V	print version and exit
  -d value
    	enable debugging settings; try -d help
  -debug
    	dump instructions as they are parsed
  -dynlink
    	support references to Go symbols defined in other shared libraries
  -e	no limit on number of errors reported
  -gensymabis
    	write symbol ABI information to output file, don't assemble
  -linkshared
    	generate code that will be linked against Go shared libraries
  -o string
    	output file; default foo.o for /a/b/c/foo.s as first argument
  -p string
    	set expected package import to path (default "<unlinkable>")
  -shared
    	generate code that can be linked into a shared library
  -spectre list
    	enable spectre mitigations in list (all, ret)
  -trimpath string
    	remove prefix from recorded source file paths
  -v	print debug output
# internal/race
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# sync/atomic
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# unicode
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# unicode/utf8
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# encoding
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/itoa
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# math/bits
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# cmp
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# unicode/utf16
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# google.golang.org/protobuf/internal/flags
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# container/list
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# crypto/internal/alias
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# crypto/subtle
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# crypto/internal/boring/sig
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# vendor/golang.org/x/crypto/cryptobyte/asn1
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/nettrace
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# vendor/golang.org/x/crypto/internal/alias
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# log/internal
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# golang.org/x/exp/constraints
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# golang.org/x/exp/maps
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# github.com/opencontainers/runc/types/features
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# github.com/opencontainers/runtime-spec/specs-go/features
compile: version "go1.22.2" does not match go tool version "go1.21.9"
gmake: *** [Makefile:71: runc-bin] Error 1



MSG:

# internal/goarch
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/unsafeheader
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/coverage/rtcov
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/cpu
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/godebugs
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/goos
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/goexperiment
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# runtime/internal/atomic
flag provided but not defined: -compiling-runtime
usage: asm [options] file.s ...
Flags:
  -D value
    	predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
    	include directory; can be set multiple times
  -S	print assembly and machine code
  -V	print version and exit
  -d value
    	enable debugging settings; try -d help
  -debug
    	dump instructions as they are parsed
  -dynlink
    	support references to Go symbols defined in other shared libraries
  -e	no limit on number of errors reported
  -gensymabis
    	write symbol ABI information to output file, don't assemble
  -linkshared
    	generate code that will be linked against Go shared libraries
  -o string
    	output file; default foo.o for /a/b/c/foo.s as first argument
  -p string
    	set expected package import to path (default "<unlinkable>")
  -shared
    	generate code that can be linked into a shared library
  -spectre list
    	enable spectre mitigations in list (all, ret)
  -trimpath string
    	remove prefix from recorded source file paths
  -v	print debug output
# runtime/internal/syscall
flag provided but not defined: -compiling-runtime
usage: asm [options] file.s ...
Flags:
  -D value
    	predefined symbol with optional simple value -D=identifier=value; can be set multiple times
  -I value
    	include directory; can be set multiple times
  -S	print assembly and machine code
  -V	print version and exit
  -d value
    	enable debugging settings; try -d help
  -debug
    	dump instructions as they are parsed
  -dynlink
    	support references to Go symbols defined in other shared libraries
  -e	no limit on number of errors reported
  -gensymabis
    	write symbol ABI information to output file, don't assemble
  -linkshared
    	generate code that will be linked against Go shared libraries
  -o string
    	output file; default foo.o for /a/b/c/foo.s as first argument
  -p string
    	set expected package import to path (default "<unlinkable>")
  -shared
    	generate code that can be linked into a shared library
  -spectre list
    	enable spectre mitigations in list (all, ret)
  -trimpath string
    	remove prefix from recorded source file paths
  -v	print debug output
# internal/race
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# sync/atomic
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# unicode
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# unicode/utf8
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# encoding
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/itoa
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# math/bits
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# cmp
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# unicode/utf16
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# google.golang.org/protobuf/internal/flags
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# container/list
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# crypto/internal/alias
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# crypto/subtle
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# crypto/internal/boring/sig
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# vendor/golang.org/x/crypto/cryptobyte/asn1
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# internal/nettrace
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# vendor/golang.org/x/crypto/internal/alias
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# log/internal
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# golang.org/x/exp/constraints
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# golang.org/x/exp/maps
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# github.com/opencontainers/runc/types/features
compile: version "go1.22.2" does not match go tool version "go1.21.9"
# github.com/opencontainers/runtime-spec/specs-go/features
compile: version "go1.22.2" does not match go tool version "go1.21.9"
gmake: *** [Makefile:71: runc-bin] Error 1

PLAY RECAP ***************************************************************************************************************************************************
localhost                  : ok=43   changed=19   unreachable=0    failed=1    skipped=7    rescued=0    ignored=0  
```

**Comment by haircommander** [comment]
you can set `GOROOT=/usr/local/go1.21.9/go` as an env var to the runc build command to fix this i think

**Comment by sohankunkerkar** [comment]
Ansible `Make` module doesn't support setting the env

File: contrib/test/ci/build/runc.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@
   make:
```
Comment (sohankunkerkar) [inline contrib/test/ci/build/runc.yml:30]:
```suggestion
  environment:
     GOROOT: "/usr/local/go1.21.9/go"
  make:
```

**Comment by kannon92** [comment]
@sohankunkerkar does this PR work now on the VM?

/hold cancel

**Comment by sohankunkerkar** [comment]
Nope, there's one more hiccup:
```sh
TASK [Run configure] *****************************************************************************************************************************************
task path: /home/skunkerk/cri-o/contrib/test/ci/build/crun.yml:59
fatal: [localhost]: FAILED! => {
    "changed": true,
    "cmd": [
        "/usr/go/src/github.com/containers/crun/configure",
        "--enable-embedded-yajl",
        "--with-wasmedge"
    ],
    "delta": "0:00:02.542671",
    "end": "2024-04-05 17:45:38.215455",
    "invocation": {
        "module_args": {
            "_raw_params": "/usr/go/src/github.com/containers/crun/configure --enable-embedded-yajl --with-wasmedge",
            "_uses_shell": false,
            "argv": null,
            "chdir": "/usr/go/src/github.com/containers/crun",
            "creates": null,
            "executable": null,
            "expand_argument_vars": true,
            "removes": null,
            "stdin": null,
            "stdin_add_newline": true,
            "strip_empty_ends": true
        }
    },
    "rc": 1,
    "start": "2024-04-05 17:45:35.672784"
}

STDOUT:

checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... gawk
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking whether make sets $(MAKE)... yes
checking whether make supports the include directive... yes (GNU style)
checking whether make supports nested variables... yes
checking whether UID '0' is supported by ustar format... yes
checking whether GID '0' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking dependency style of gcc... gcc3
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports nested variables... (cached) yes
checking for gawk... (cached) gawk
checking for a sed that does not truncate output... (cached) /usr/bin/sed
checking for gcc... (cached) gcc
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to enable C11 features... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking for a Python interpreter with version >= 3... python
checking for python... /usr/bin/python
checking for python version... 3.10
checking for python platform... linux
checking for GNU default python prefix... ${prefix}
checking for GNU default python exec_prefix... ${exec_prefix}
checking for python script directory (pythondir)... ${PYTHON_PREFIX}/lib/python3.10/site-packages
checking for python extension module directory (pyexecdir)... ${PYTHON_EXEC_PREFIX}/lib64/python3.10/site-packages
checking for go-md2man... /usr/bin/go-md2man
checking for error.h... yes
checking for linux/openat2.h... yes
checking for stdatomic.h... yes
checking for linux/ioprio.h... yes
checking for atomic_int... yes
checking for copy_file_range... yes
checking for fgetxattr... yes
checking for statx... yes
checking for fgetpwent_r... yes
checking for issetugid... no
checking for memfd_create... yes
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for sys/capability.h... yes
checking for library containing cap_from_name... -lcap
checking for library containing dlopen... none required
checking for wasmedge/wasmedge.h... no


STDERR:

configure: error: *** Missing wasmedge headers


MSG:

non-zero return code

PLAY RECAP ***************************************************************************************************************************************************
localhost                  : ok=51   changed=19   unreachable=0    failed=1    skipped=10   rescued=0    ignored=0  
```
/hold

Still testing that part

**Comment by sohankunkerkar** [comment]
hmm... it looks like you need to revert [this](https://github.com/cri-o/cri-o/pull/7951) change to address the above issue.

**Comment by kannon92** [comment]
> hmm... it looks like you need to revert [this](https://github.com/cri-o/cri-o/pull/7951) change to address the above issue.

I added the revert to this PR.

**Comment by kannon92** [comment]
With @sohankunkerkar lgtm, going to remove hold.

/hold cancel

will monitor tomorrow and see if VM builds are good!

**Pull request merged**